src/App.tsx

```diff
@@ -1,24 +1,239 @@
+    .filter(({ completed }) => completed === false).length;
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingId(prevTodoIds => [...prevTodoIds, todoId]);
+    setIsLoaderActive(true);
+
+    return postService.deleteTodo(todoId)
+      .then(() => {
+        setTodoList(currentList => currentList
+          .filter(todo => todo.id !== todoId));
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.DeleteError);
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    completedTodosId
+      .forEach((id) => {
+        handleDeleteTodo(id);
+      });
+  };
+
+  const addNewTodo = (todo: Todo) => {
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    });
+
+    setErrorMessage(ErrorMessages.NoError);
+    setLoadingId([0]);
+    setIsLoaderActive(true);
+
+    postService.createTodo(todo)
+      .then(newTodo => {
+        setTodoList(currentTodos => [...currentTodos, newTodo]);
+        setTitle('');
+        setTimeout(() => {
+          setIsLoaderActive(false);
+          setLoadingId([]);
+        }, 3000);
+      })
+      .catch(() => {
+        setIsLoaderActive(false);
+        setLoadingId([0]);
+        setErrorMessage(ErrorMessages.AddError);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setLoadingId([]);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const updateTodo = (todoToUpdate: Todo) => {
+    return postService.updateTodo(todoToUpdate)
+      .then(todo => {
+        setTodoList(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(({ id }) => id === todo.id);
+
+          newTodos.splice(index, 1, todo);
+
+          return newTodos;
+        });
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.UpdateError);
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setErrorMessage(ErrorMessages.EmptyTitleError);
+
+      return;
+    }
+
+    addNewTodo({
+      id: +new Date(),
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    });
+  };
+
+  const handleComplete = (updatedTodo: Todo) => {
+    setErrorMessage(ErrorMessages.NoError);
+    setLoadingId([updatedTodo.id]);
+    setIsLoaderActive(true);
+
+    const foundTodo = { ...updatedTodo };
+
+    foundTodo.completed = !foundTodo.completed;
+
+    updateTodo(foundTodo);
+  };
+
+  const handleToggleAll = () => {
+    setIsToggled(prev => !prev);
+    setIsLoaderActive(true);
+    setLoadingId([...allTodosId]);
+
+    todoList.map(todo => {
+      const item = { ...todo, completed: !isToggled };
+
+      updateTodo(item);
+
+      return item;
+    });
+  };
```

Comment:
When a user has uncompleted todos they should be marked as completed, in your app it works in the opposite way.

The flow should look something like this:
1) If you have uncompleted todos, filter only them. If not all todos should be changed.
2) Map todos for update array, for each iteration toggle status, and send the request to API.
3) With `Promise.all` check the response, if all is good set updates, if an error occurs notify the user with an error message
4) In finally block remove loaders from these todos